lib/launchers/ocm.rb

```diff
@@ -0,0 +1,197 @@
+    # in the future we plan to move the logic into the script 'osd-provision.sh'
+    def generate_json(name)
+      json_data = {
+        "name" => name,
+        "managed" => true,
+        "multi_az" => false,
+        "byoc" => false
+      }
+
+      if @multi_az
+        json_data.merge!({"multi_az" => @multi_az})
+      end
+
+      if @region
+        json_data.merge!({"region" => {"id" => @region}})
+      end
+
+      if @version
+        json_data.merge!({"version" => {"id" => "openshift-v#{@version}"}})
+      end
+
+      if @nodes
+        json_data.merge!({"nodes" => {"compute" => @nodes.to_i}})
+      end
+
+      if @lifespan
+        expiration = Time.now + to_seconds(@lifespan)
+        json_data.merge!({"expiration_timestamp" => expiration.strftime("%Y-%m-%dT%H:%M:%SZ")})
+      end
+
+      if @cloud_opts
+        case @cloud_opts[:cloud_type]
+        when "aws"
+          aws = Amz_EC2.new(service_name: @cloud)
+          json_data.merge!({"aws" => {"access_key_id":aws.access_key, "secret_access_key":aws.secret_key, "account_id":aws.account_id}})
+          json_data.merge!({"byoc" => true})
+        end
+      end
+
+      return json_data.to_json
+    end
+
+    # download the script 'osd-provision.sh' which takes care of the OSD installation/uninstallation
+    def download_osd_script
+      default_osd_script_url = "https://gitlab.cee.redhat.com/mk-bin-packing/mk-performance-tests/-/raw/master/scripts/osd-provision.sh?inline=false"
+      osd_script_url = ENV['OSD_SCRIPT_URL'] || default_osd_script_url
+      # we need another parent dir as the script downloads ocm tool to its parent
+      osd_scripts = "#{Dir.tmpdir}/osd-scripts"
+      %x(
+        rm -rf #{osd_scripts} && mkdir #{osd_scripts} && \
+        curl -s #{osd_script_url} --output #{osd_scripts}/osd-provision.sh && \
+        chmod a+x #{osd_scripts}/osd-provision.sh
+      )
+      return "#{osd_scripts}/osd-provision.sh"
+    end
+
+    def shell(cmd)
+      if !system(cmd)
+        raise "Error when executing '#{cmd}'"
+      end
+    end
+
+    # create OSD cluster
+    def create_osd(name)
+      # cerate a temp file with ocm-token
+      ocm_token_file = Tempfile.new("ocm-token-file", Host.localhost.workdir)
+      File.open(ocm_token_file, "w") do |f|
+        f.write(@token)
+      end
+      # create a temp file with cluster specification
+      puts File.read(ocm_token_file.path)
+      ocm_json_file = Tempfile.new("ocm-json-file", Host.localhost.workdir)
+      File.open(ocm_json_file, "w") do |f|
+        data = generate_json(name)
+        f.write(data)
+      end
+      # now, download the script which will take care of the OSD cluster installation
+      osd_script = download_osd_script
+      shell("#{osd_script} --create --cloud-token-file #{ocm_token_file.path} -f #{ocm_json_file.path} --wait")
+      shell("#{osd_script} --get api_url -f #{ocm_json_file.path}")
+      shell("#{osd_script} --get credentials -f #{ocm_json_file.path}")
```

Comment:
I see we can't avoid using shell commands given we depend on a script actually. But lets limit local command execution to this script only. Just a note, not request to change.

Reply:
ok